### ISSUE TITLE
refference

### ISSUE THREAD

**Issue by EverNife** [tterrag1098/CustomThings#35]
I am new to this mod, and i want to create a few itens, but i can't find any good exemple of each thing.

For exemple, i want to create a block and there is no explanation for where to put it's texture (at least i didn't find it )

So please, can you give me a functional ZIP file with an exemple of each thing?

(sorry for bad english '-' )

**Comment by tterrag1098** [comment]
Block icons go in /icons with everything else. I think the wiki is more than enough. If you find anything to be missing let me know.

**Comment by EverNife** [comment]
Ok, i will, 
Keep this open a bit more please.

To start, the [Custom ITEM](https://github.com/tterrag1098/CustomThings/wiki/Custom-Items) exemple is broken.

It is reporting [this crash](http://pastebin.com/jTMtEqvJ).

Can you give me a funcional one?

(Sorry again, i don't know json language )

**Comment by tterrag1098** [comment]
Yes, it was incorrect. I have updated it now.

**Comment by EverNife** [comment]
Lang tutorial is wrong too, need the "tile." before the expression.

In [the wiki](https://github.com/tterrag1098/CustomThings/wiki/Adding-Icons-and-Lang-Files) is: 

> item.itemname.name=My Item

it should be:

> tile.customthings.itemname.name=My Item

**Comment by EverNife** [comment]
Else

Getting [this crash](http://pastebin.com/KiS72HfF) while trying to make an amor.

This is the [customArmors.json](http://pastebin.com/CsG1C7XZ)

**Comment by EverNife** [comment]
Can  you tell me what have i done wrong in the customarmors.json?

**Comment by tterrag1098** [comment]
You have an extra comma on line 11

**Comment by EverNife** [comment]
'-' facepalm '-' My bad.

I took liberty to edit your wiki [Custom Armor](https://github.com/tterrag1098/CustomThings/wiki/Custom-Armor) and put some examples. 
I will need to keep this open to take more of my doubts out.

**Comment by EverNife** [comment]
[Custom Items](https://github.com/tterrag1098/CustomThings/wiki/Custom-Items) exemple is not working :/ 

Getting this error:  [PasteBin](http://pastebin.com/0JXv5PW6)

**Comment by EverNife** [comment]
Help please ^^ 

(sorry for being so boring)

**Comment by xpto2011** [comment]
Also getting the same crash when trying to create the example item

**Comment by EverNife** [comment]
Life is hard with us all T.T

**Comment by tterrag1098** [comment]
Could you be more specific? What crash? What (_exactly_) is your json?

**Comment by EverNife** [comment]
> evernife commented on 8 Jul
> 
> [Custom Item](https://github.com/tterrag1098/CustomThings/wiki/Custom-Items) exemple is not working :/
> Getting this error: [PasteBin](http://pastebin.com/0JXv5PW6)

**Comment by xpto2011** [comment]
I get it working by only setting the name parameter.

The json I used before was exactly the one in the wiki and it crashed with this crashreport: http://pastebin.com/br0pkndL

**Comment by evernife** [comment]
ty :D

**Comment by EverNife** [comment]
Texture for tools no working well `(or it is?)`

The wiki tell us to just put the **_NameOfTheItem.png**_ in the icon's folder.

But in the case of tools, you need to put the type of the tool too.

For exemple, creating a sword named `spectrum` i need to put the name of the icon as `spectrumSword.png`
Same for pickaxe, one called `dilmae_is_more` need a png called `dilmae_is_morePickaxe`

And for the case of multiples types seems to take ther first one.

Is this intentional?

**Comment by tterrag1098** [comment]
Yes that is the correct naming scheme for tools. I can add that to the wiki soon. What is the problem though?

**Comment by EverNife** [comment]
Relative to refference issue;

Cause its a bit strange, my items are now called, for exemple `sword_spectrum` => `sword_spectrumSword.png`

Not a poblem at all ^^

**Comment by DarkBlade177** [comment]
@tterrag1098 how can i set up block textureMap ?

I tried

```
   "textureMap" : [
    {"derp1"},
    {"derp1"},
    {"derp2"},
    {"derp2"},
    {"derp2"},
    {"derp2"}
  ]
```

But did not work T.T

**Comment by DarkBlade177** [comment]
This way should work:

		"textureMap": ["derp1","derp1","derp2","derp2","derp2","derp2"]

**Comment by DarkBlade177** [comment]
This way worked for me, forget it:

```
    "textureMap": ["derp1","derp1","derp2","derp2","derp2","derp2"]
```
